temporal-sdk/src/main/java/io/temporal/internal/testservice/StateMachines.java

```diff
@@ -199,20 +199,23 @@
     String originalExecutionRunId;
     Optional<String> continuedExecutionRunId;
     Functions.Proc runTimerCancellationHandle;
+    Failure lastFailure;
 
     WorkflowData(
         Optional<TestServiceRetryState> retryState,
         Duration backoffStartInterval,
         String cronSchedule,
         Payloads lastCompletionResult,
         String originalExecutionRunId,
-        Optional<String> continuedExecutionRunId) {
+        Optional<String> continuedExecutionRunId,
+        Failure lastFailure) {
```

Comment:
Shouldn't this be Optional?

Reply:
Could be, for sure. Java is killing me a bit with the whole `Optional<T>` actually has three states problem, where it can still be null. IntelliJ has a lint that whines at you when you use it as a parameter, which I've always found a bit odd, but there are reasons: https://stackoverflow.com/questions/31922866/why-should-java-8s-optional-not-be-used-in-arguments

I don't really have much of a preference. Since we're already using them as arguments here though, I should probably stay consistent. Curious to hear your thoughts though @mfateev